src/errors/UndefinedProtectedPrivateKeyError.ts

```diff
@@ -0,0 +1,8 @@
+class UndefinedProtectedPrivateKeyError extends Error {
+  constructor(msg: string) {
+    super(msg);
+    Object.setPrototypeOf(this, UndefinedProtectedPrivateKeyError.prototype);
+  }
+}
+
+export default UndefinedProtectedPrivateKeyError;
```

Comment:
Can we use function component instead class component? https://ko.reactjs.org/docs/components-and-props.html#function-and-class-components

Reply:
It seems an inheritance of `Error` class, not `React.Component`.

Reply:
Oh yes 😅  I misreading class name lol